+nosnoc/+discrete_time_problem/Gcs.m

```diff
@@ -0,0 +1,870 @@
+                            if size(model.G_path, 1) > 0
+                                G_path = dcs.G_path_fun(x_ijk, z_ijk, ui, v_global, p);
+                                H_path = dcs.H_path_fun(x_ijk, z_ijk, ui, v_global, p);
+                                obj.G.path(ii,jj,kk) = {G_path};
+                                obj.H.path(ii,jj,kk) = {H_path};
+                            end
+                            if opts.cost_integration
+                                % also integrate the objective
+                                obj.f = obj.f + opts.b_rk(kk)*h*qj;
+                            end
+                        end
+                        if ~opts.right_boundary_point_explicit
+                            x_ijk = obj.w.x(ii,jj,opts.n_s+1);
+                            z_ijk = obj.w.z(ii,jj,opts.n_s+1);
+                            lambda_ijk = obj.w.lambda(ii,jj,opts.n_s+1);
+
+                            obj.g.dynamics(ii,jj,opts.n_s+1) = {x_ijk - x_ij_end};
+                            obj.g.c_lb(ii,jj,opts.n_s+1) = {dcs.c_fun(x_ijk, z_ijk, v_global, p), 0, inf};
+                            obj.g.z(ii,jj,opts.n_s+1) = {dcs.g_z_fun(x_ijk, z_ijk, ui, v_global, p)};
+                        else
+                            obj.g.dynamics(ii,jj,opts.n_s+1) = {x_ij_end - obj.w.x(ii,jj,opts.n_s)};
+                        end
+                        if ~opts.g_path_at_stg && opts.g_path_at_fe
+                            obj.g.path(ii,jj) = {dcs.g_path_fun(x_ijk, z_ijk, ui, v_global, p), model.lbg_path, model.ubg_path};
+                        end
+                    end
+                    x_prev = obj.w.x(ii,jj,opts.n_s+rbp);
+                end
+                if ~opts.g_path_at_stg && ~opts.g_path_at_fe
+                    x_i = obj.w.x(ii, opts.N_finite_elements(ii), opts.n_s);
+                    z_i = obj.w.z(ii, opts.N_finite_elements(ii), opts.n_s);
+                    obj.g.path(ii) = {dcs.g_path_fun(x_i, z_i, ui, v_global, p), model.lbg_path, model.ubg_path};
+                end
+
+                % Least Squares Costs
+                % TODO we should convert the refs to params
+                if ~isempty(model.x_ref_val)
+                    obj.f = obj.f + t_stage*dcs.f_lsq_x_fun(obj.w.x(ii,opts.N_finite_elements(ii),opts.n_s),...
+                        model.x_ref_val(:,ii),...
+                        p);
+                end
+                if ~isempty(model.u_ref_val)
+                    obj.f = obj.f + t_stage*dcs.f_lsq_u_fun(obj.w.u(ii),...
+                        model.u_ref_val(:,ii),...
+                        p);
+                end
+                if ~opts.cost_integration
+                    obj.f = obj.f + dcs.f_q_fun(x_ijk, z_ijk, lambda_ijk, ui, v_global, p);
+                end
+
+                % Clock Constraints
+                % TODO(@anton) HERE BE DRAGONS. This is by far the worst part of current nosnoc as it requires the discrete problem
```

Comment:
this comment here is irrelevant. we dont have time freezing for PDS (and first order systems in general, as they dont have state jumps do to constraint, in order to be meaningful)